packages/admin/admin-rte/src/core/Controls/ControlButton.tsx

```diff
@@ -1,11 +1,71 @@
-import { ComponentsOverrides, Theme } from "@mui/material";
+import { ThemedComponentBaseProps } from "@comet/admin";
+import { ComponentsOverrides } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
 import { SvgIconProps } from "@mui/material/SvgIcon";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
 import * as React from "react";
 
 import getRteTheme from "../utils/getRteTheme";
 
-export interface IProps {
+export type RteControlButtonClassKey = "root" | "selected" | "renderAsIcon";
+
+type OwnerState = Pick<IProps, "selected" | "Icon">;
+
+const Root = styled("button", {
+    name: "CometAdminRteControlButton",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [styles.root, ownerState.selected && styles.selected, ownerState.Icon && styles.renderAsIcon];
+    },
+})<{ ownerState: OwnerState }>(
+    ({ ownerState, theme }) => css`
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        cursor: pointer;
+        height: 24px;
+        background-color: transparent;
+        border: 1px solid transparent;
+        box-sizing: border-box;
+        transition: background-color 200ms, border-color 200ms, color 200ms;
+        font-size: 20px;
+        color: ${getRteTheme(theme.components?.CometAdminRte?.defaultProps).colors?.buttonIcon};
+
+        &:hover {
+            background-color: ${getRteTheme(theme.components?.CometAdminRte?.defaultProps).colors?.buttonBackgroundHover};
+            border-color: ${getRteTheme(theme.components?.CometAdminRte?.defaultProps).colors?.buttonBorderHover};
+        }
+
+        &:disabled {
+            cursor: not-allowed;
+
+            &,
+            &:hover {
+                background-color: transparent;
+                border-color: transparent;
+                color: ${getRteTheme(theme.components?.CometAdminRte?.defaultProps).colors?.buttonIconDisabled};
+            }
+        }
+
+        ${ownerState.selected &&
+        css`
+            &:not(:disabled),
+            &:not(:disabled):hover {
+                border-color: ${getRteTheme(theme.components?.CometAdminRte?.defaultProps).colors?.buttonBorderHover};
+                background-color: white;
+            }
+        `}
+
+        ${ownerState.Icon &&
```

Comment:
Please only use boolean-checks with the `&&` syntax:

```suggestion
        ${Boolean(ownerState.Icon) &&
```